morpheus/_lib/tests/stages/test_preprocess_fil.cpp

```diff
@@ -19,35 +36,51 @@ TEST_F(TestPreprocessFIL, TestProcessControlMessageAndMultiMessage)
 
     // Create ControlMessage
     auto cm = std::make_shared<ControlMessage>();
-    cm->payload(meta);
+    cm->payload(cm_meta);
 
     // Create PreProcessControlMessageStage
-    auto cm_stage =
-        std::make_shared<PreprocessFILStage<ControlMessage, ControlMessage>>(std::vector<std::string>{"country"});
-
-    auto cm_response         = cm_stage->pre_process_batch(cm);
-    auto cm_response_payload = cm_response->payload();
-    EXPECT_EQ(cm_response_payload->count(), 193);
+    auto cm_stage    = std::make_shared<PreprocessFILStageCC>(std::vector<std::string>{"float_str1", "float_str2"});
+    auto cm_response = cm_stage->on_data(cm);
 
     // Create MultiMessage
-    auto multi = std::make_shared<MultiMessage>(meta);
+    auto mm = std::make_shared<MultiMessage>(mm_meta);
     // Create PreProcessMultiMessageStage
-    auto multi_stage =
-        std::make_shared<PreprocessFILStage<MultiMessage, MultiInferenceMessage>>(std::vector<std::string>{"country"});
-    auto multi_response         = multi_stage->pre_process_batch(multi);
-    auto multi_response_payload = multi_response->meta;
+    auto mm_stage    = std::make_shared<PreprocessFILStageMM>(std::vector<std::string>{"float_str1", "float_str2"});
+    auto mm_response = mm_stage->on_data(mm);
+
+    auto cm_tensors = cm_response->tensors();
+    auto mm_tensors = mm_response->memory;
+
+    // Verify output tensors
+    std::vector<float> expected_input__0 = {1, 4, 2, 5, 3, 6};
+    auto cm_input__0                     = cm_tensors->get_tensor("input__0");
+    auto mm_input__0                     = mm_tensors->get_tensor("input__0");
+    std::vector<float> cm_input__0_host(cm_input__0.count());
+    std::vector<float> mm_input__0_host(mm_input__0.count());
+    MRC_CHECK_CUDA(cudaMemcpy(
+        cm_input__0_host.data(), cm_input__0.data(), cm_input__0.count() * sizeof(float), cudaMemcpyDeviceToHost));
+    MRC_CHECK_CUDA(cudaMemcpy(
+        mm_input__0_host.data(), mm_input__0.data(), mm_input__0.count() * sizeof(float), cudaMemcpyDeviceToHost));
+    EXPECT_EQ(expected_input__0, cm_input__0_host);
+    EXPECT_EQ(cm_input__0_host, mm_input__0_host);
 
-    // Check if identical number of rows are returned
-    EXPECT_EQ(multi_response_payload->count(), cm_response_payload->count());
+    // Col1 in MatxUtil__MatxCreateSegIds is not initialized
+    // auto cm_seq_ids = cm_tensors->get_tensor("seq_ids");
+    // auto mm_seq_ids = mm_tensors->get_tensor("seq_ids");
+    // std::vector<TensorIndex> cm_seq_ids_host(cm_seq_ids.count());
+    // std::vector<TensorIndex> mm_seq_ids_host(mm_seq_ids.count());
+    // MRC_CHECK_CUDA(cudaMemcpy(cm_seq_ids_host.data(), cm_seq_ids.data(), cm_seq_ids.count() * sizeof(TensorIndex),
+    // cudaMemcpyDeviceToHost)); MRC_CHECK_CUDA(cudaMemcpy(mm_seq_ids_host.data(), mm_seq_ids.data(), mm_seq_ids.count()
+    // * sizeof(TensorIndex), cudaMemcpyDeviceToHost)); EXPECT_EQ(cm_seq_ids_host, mm_seq_ids_host);
```

Comment:
To enable this check to function, update `MatxUtil__MatxCreateSegIds` to initialize column 1 to 0 so we can consistently check the output without uninitialized values.